docs/rst/tutorials/core/deployment/dds2vulcanexus/dds2vulcanexus_topic.rst

```diff
@@ -0,0 +1,312 @@
+
+.. _dds2vulcanexus_topic_fastdds_generation:
+
+Fast DDS Application - Type generation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the case of Fast DDS, the type support is generated from the `HelloWorld.idl` file using `*Fast DDS-Gen* <https://fast-dds.docs.eprosima.com/en/latest/fastddsgen/introduction/introduction.html>`_.
+
+In this tutorial, the Fast DDS type support is generated within the `CMakeLists.txt` file for the sake of completion and simplicity, but it can be generated as a pre-build step instead.
+Inspecting the `CMakeLists.txt` file downloaded in :ref:`dds2vulcanexus_topic_prerequisites`, the following CMake code pertains the native Fast DDS subscriber:
+
+.. literalinclude:: /resources/tutorials/core/deployment/dds2vulcanexus/topic/CMakeLists.txt
+    :language: CMake
+    :lines: 59-93
+
+In particular, the type generation related code is:
+
+.. literalinclude:: /resources/tutorials/core/deployment/dds2vulcanexus/topic/CMakeLists.txt
+    :language: CMake
+    :lines: 65-82
+
+The call to *Fast DDS-Gen* within ``add_custom_command`` will generate the type support in the `fastdds_app` directory, leaving the file names in a convenient ``GENERATED_TYPE_SUPPORT_FILES`` CMake variable that is later used to add the source files to the executable.
+It is important to note the *Fast DDS-Gen* is called with the `-typeros2` flag, so it generates ROS 2 compatible type names.
+
+.. _dds2vulcanexus_topic_fastdds_c++:
+
+Fast DDS Application - C++
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The native Fast DDS subscriber is as follows:
+
+.. literalinclude:: /resources/tutorials/core/deployment/dds2vulcanexus/topic/fastdds_app/subscriber.cpp
+    :language: c++
+
+There are several things to unpack in this application:
+
+1. ``HelloWorldSubscriber`` holds both a reference to the type support, to create the topic, and a ``HelloWorld`` sample instance for reusing it upon reception.
+
+    .. literalinclude:: /resources/tutorials/core/deployment/dds2vulcanexus/topic/fastdds_app/subscriber.cpp
+        :language: c++
+        :lines: 115-116
+        :dedent: 4
+
+    1. The type support is instantiated upon construction:
+
+        .. literalinclude:: /resources/tutorials/core/deployment/dds2vulcanexus/topic/fastdds_app/subscriber.cpp
+            :language: c++
+            :lines: 37-42
+            :dedent: 4
+
+    2. Then, it is registered in the ``DomainParticipant`` for further use:
+
+        .. literalinclude:: /resources/tutorials/core/deployment/dds2vulcanexus/topic/fastdds_app/subscriber.cpp
+            :language: c++
+            :lines: 72
+            :dedent: 8
+
+2. The topic is created with name ``rt/HelloWorld``.
+   Mind that this topic name is different from the one set in the Vulcanexus publisher (``HelloWorld``).
+   This is because Vulcanexus appends ``rt/`` to the topic name passed when creating a ``Publisher`` or ``Subscription``, where ``rt`` stands for *ROS Topic*, as services and actions have different prefixes.
```

Comment:
Adding a link to https://design.ros2.org/articles/topic_and_service_names.html may be helpful.